tests/rdf_tests/rdflib_tools.py

```diff
@@ -0,0 +1,201 @@
+    TestEval: URIRef  # Superclass of all RDF Evaluation Tests.
+    TestNQuadsNegativeSyntax: URIRef  # A negative N-Quads syntax test.
+    TestNQuadsPositiveSyntax: URIRef  # A positive N-Quads syntax test.
+    TestNTriplesNegativeSyntax: URIRef  # A negative N-Triples syntax test.
+    TestNTriplesPositiveSyntax: URIRef  # A positive N-Triples syntax test.
+    TestSyntax: URIRef  # Superclass of all RDF Syntax Tests.
+    TestTrigEval: URIRef  # A positive TriG evaluation test.
+    TestTrigNegativeEval: URIRef  # A negative TriG evaluation test.
+    TestTriGNegativeSyntax: URIRef  # A negative TriG syntax test.
+    TestTriGPositiveSyntax: URIRef  # A positive TriG syntax test.
+    TestTurtleEval: URIRef  # A positive Turtle evaluation test.
+    TestTurtleNegativeEval: URIRef  # A negative Turtle evaluation test.
+    TestTurtleNegativeSyntax: URIRef  # A negative Turtle syntax test.
+    TestTurtlePositiveSyntax: URIRef  # A positive Turtle syntax test.
+    Test: URIRef  # Superclass of all RDF Tests.
+    TestXMLNegativeSyntax: URIRef  # A negative RDF/XML syntax test.
+    XMLEval: URIRef  # A positive RDF/XML evaluation test.
+
+    TestTrigPositiveSyntax: URIRef
+    TestTrigNegativeSyntax: URIRef
+
+
+DAWG = Namespace("http://www.w3.org/2001/sw/DataAccess/tests/test-dawg#")
+
+ResultType = Union[Identifier, Tuple[Identifier, List[Tuple[Identifier, Identifier]]]]
+GraphDataType = Union[List[Identifier], List[Tuple[Identifier, Identifier]]]
+
+
+class RDFTest(NamedTuple):
+    uri: URIRef
+    name: str
+    comment: str
+    data: Identifier
+    graphdata: Optional[GraphDataType]
+    action: Identifier
+    result: Optional[ResultType]
+    syntax: bool
+
+
+def read_manifest(f, base=None, legacy=False) -> Iterable[Tuple[Node, URIRef, RDFTest]]:
+    def _str(x):
+        if x is not None:
+            return str(x)
+        return None
+
+    g = Graph()
+    g.parse(f, publicID=base, format="turtle")
+
+    for m in g.subjects(RDF.type, MF.Manifest):
+
+        for col in g.objects(m, MF.include):
+            for i in g.items(col):
+                for x in read_manifest(i):
+                    yield x
+
+        for col in g.objects(m, MF.entries):
+            e: URIRef
+            for e in g.items(col):
+
+                approved = (
+                    (e, DAWG.approval, DAWG.Approved) in g
+                    or (e, DAWG.approval, DAWG.NotClassified) in g
+                    or (e, RDFT.approval, RDFT.Approved) in g
+                )
+
+                # run proposed tests
+                # approved |= (e, RDFT.approval, RDFT.Proposed) in g
+
+                # run legacy tests with no approval set
+                if legacy:
+                    approved |= (e, DAWG.approval, None) not in g and (
+                        e,
+                        RDFT.approval,
+                        None,
+                    ) not in g
+
+                if not approved:
```

Comment:
Oxigraph tests are not from the W3C working group so the DAWG.approval property is not set, leading to `approved = False`. This explains why Oxigraph tests are ignored.